passes/src/Luna/Pass/Evaluation/Interpreter.hs

```diff
@@ -241,11 +241,15 @@ interpret' glob expr = Layer.read @IR.Model expr >>= \case
             lift $ State.evalT cl (Scope.merge scope env)
     Uni.Marked _ b' -> do
         b <- IR.source b'
+        interpretHere <- Layer.read @IR.Model b >>= \case
+            Uni.Unify{}    -> return False
+            Uni.Function{} -> return False
+            a              -> return True
         e <- interpret glob b
-        return $ do
+        return $ if not interpretHere then e else do
```

Comment:
I wonder whether using `interpret glob b >>= \e -> if [...]` wouldn't be more readable.